src/Compilers/Core/MSBuildTaskTests/CscTests.cs

```diff
@@ -27,6 +27,15 @@ public void MultipleSourceFiles()
             Assert.Equal("/out:test1.exe test1.cs test2.cs", csc.GenerateResponseFileContents());
         }
 
+        [Fact] 
+        public void DeterministicFlag()
+        {
+            var csc = new Csc();
+            csc.Sources = MSBuildUtil.CreateTaskItems("test.cs");
+            csc.Deterministic = true;
+            Assert.Equal("/deterministic+ /out:test.exe test.cs", csc.GenerateResponseFileContents());
+        }
```

Comment:
@tmat What do you mean by a combination of them?  There shouldn't be a way for it to generate `/deterministic-` and `/deterministic+` for the same build.